File: py/Majority_Element.py
class Solution:
    def majorityElement(self, nums):
        res, cnt, l = nums[0], 1, len(nums)
        for i in range(1, l):
            if res == nums[i]:
                cnt += 1
            elif cnt > 0:
                cnt -= 1
            else:
                res = nums[i]
                cnt += 1
        return res

File: c_plus/Majority_Element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int res = nums[0];
        int cnt = 1;
        int l = nums.size();
        for (int i = 1; i < l; i++) {
            if (res == nums[i]) cnt++;
            else if (cnt > 0) cnt--;
            else {
                res = nums[i];
                cnt += 1;
            }
        }
        return res;
    }
};